### PR TITLE
Add Metric icon to Ranking Filter when enableRenamingMeasureToMetric FF turns on.

### PR DESCRIPTION
**Pull request by dlcd2203**
JIRA: SD-1987

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [x] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [x] `check` passes
-   [x] `check-extended` passes
-   [x] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6335/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8143/ : SUCCESS in 21m 51s

**Comment by dlcd2203** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1293/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6336/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1293/ : UNSTABLE in 20m 14s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6341/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1295/ : SUCCESS in 17m 46s

File: libs/sdk-ui-filters/api/sdk-ui-filters.api.md [hunk 1]
```diff
@@ -426,6 +426,8 @@ export interface IRankingFilterDropdownProps {
     // (undocumented)
     customGranularitySelection?: ICustomGranularitySelection;
     // (undocumented)
+    enableRenamingMeasureToMetric?: boolean;
```
Comment (anvuong119) [inline libs/sdk-ui-filters/api/sdk-ui-filters.api.md:429]: Just a confirmation: should we add `enableRenamingMeasureToMetric?: boolean;` to `IRankingFilterProps` at https://github.com/gooddata/gooddata-ui-sdk/blob/e8018b771f93a9c0560dcb3d8c2e4a1771468728/libs/sdk-ui-filters/src/RankingFilter/RankingFilter.tsx#L12

and then pass the new prop down the line to `RankingFilterDropdown` component at https://github.com/gooddata/gooddata-ui-sdk/blob/e8018b771f93a9c0560dcb3d8c2e4a1771468728/libs/sdk-ui-filters/src/RankingFilter/RankingFilter.tsx#L65

As far as I've searched, it seems that `RankingFilter` component from the SDK is currently not used anywhere. However, adding an optional prop to it should not cause any harm. In addition, in the future if the component is used, the prop would be ready to be used.

File: libs/sdk-ui-kit/styles/icons/metric.svg [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+<path d="M5.39888 8.6196C5.54621 8.47228 5.54621 8.23341 5.39888 8.08608C5.25156 7.93875 5.01269 7.93875 4.86536 8.08608L3.98471 8.96673L3.10415 8.08617C2.95682 7.93884 2.71796 7.93884 2.57063 8.08617C2.4233 8.2335 2.4233 8.47236 2.57063 8.61969L3.45119 9.50025L2.57046 10.381C2.42313 10.5283 2.42313 10.7672 2.57046 10.9145C2.71779 11.0618 2.95665 11.0618 3.10398 10.9145L3.98471 10.0338L4.86553 10.9146C5.01286 11.0619 5.25173 11.0619 5.39906 10.9146C5.54639 10.7673 5.54639 10.5284 5.39906 10.3811L4.51824 9.50025L5.39888 8.6196Z" fill="#00C18D"/>
+<path fill-rule="evenodd" clip-rule="evenodd" d="M3 0C1.34315 0 0 1.34315 0 3V11C0 12.6569 1.34315 14 3 14H11C12.6569 14 14 12.6569 14 11V3C14 1.34315 12.6569 0 11 0H3ZM11 0.75H3C1.75736 0.75 0.75 1.75736 0.75 3V11C0.75 12.2426 1.75736 13.25 3 13.25H11C12.2426 13.25 13.25 12.2426 13.25 11V3C13.25 1.75736 12.2426 0.75 11 0.75Z" fill="#00C18D"/>
+</svg>
```
[the remark this answers is not in the document]
Comment (dlcd2203) [reply libs/sdk-ui-kit/styles/icons/metric.svg]: fixed

File: libs/sdk-ui-filters/api/sdk-ui-filters.api.md (hunk 1 above)
Comment (dlcd2203) [reply libs/sdk-ui-filters/api/sdk-ui-filters.api.md:429]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6342/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8149/ : SUCCESS in 16m 50s

**Comment by dlcd2203** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**dateFilter.spec.ts**|7|-|13.4870s|
|✅|**drilling.spec.ts**|2|-|9.6920s|
|✅|**dashboard.spec.ts**|9|-|13.8040s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/96/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6343/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/96/ : SUCCESS in 8m 10s

File: libs/sdk-ui-filters/src/RankingFilter/MeasureDropdown/MeasureDropdownBody.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ interface IMeasureDropdownBodyProps {
     onClose: () => void;
```
Comment (anvuong119) [inline libs/sdk-ui-filters/src/RankingFilter/MeasureDropdown/MeasureDropdownBody.tsx:12]: Please update the copyright of this file as well. As we are making some changes now, the copyright should change accordingly, shouldn't it?

`// (C) 2020 GoodData Corporation`

↓

`// (C) 2020-2022 GoodData Corporation`

File: libs/sdk-ui-filters/src/RankingFilter/MeasureDropdown/MeasureDropdownItem.tsx [hunk 4]
```diff
@@ -11,6 +11,7 @@ interface IMeasureDropdownItemProps {
     onSelect: (ref: ObjRefInScope) => void;
```
Comment (anvuong119) [inline libs/sdk-ui-filters/src/RankingFilter/MeasureDropdown/MeasureDropdownItem.tsx:11]: Please update the copyright of this file as well. As we are making some changes now, the copyright should change accordingly, shouldn't it?

`// (C) 2020 GoodData Corporation`

↓

`// (C) 2020-2022 GoodData Corporation`

File: libs/sdk-ui-filters/src/RankingFilter/RankingFilter.tsx [hunk 5]
```diff
@@ -20,6 +20,7 @@ export interface IRankingFilterProps {
     onDropDownItemMouseOut?: () => void;
```
Comment (anvuong119) [inline libs/sdk-ui-filters/src/RankingFilter/RankingFilter.tsx:20]: Please update the copyright of this file as well. As we are making some changes now, the copyright should change accordingly, shouldn't it?

`// (C) 2020 GoodData Corporation`

↓

`// (C) 2020-2022 GoodData Corporation`

File: libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdown.tsx [hunk 6]
```diff
@@ -40,6 +40,7 @@ export interface IRankingFilterDropdownProps {
     anchorEl?: HTMLElement | string;
```
Comment (anvuong119) [inline libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdown.tsx:40]: Please update the copyright of this file as well. As we are making some changes now, the copyright should change accordingly, shouldn't it?

`// (C) 2020 GoodData Corporation`

↓

`// (C) 2020-2022 GoodData Corporation`

File: libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdownBody.tsx [hunk 7]
```diff
@@ -37,6 +37,7 @@ interface IRankingFilterDropdownBodyComponentOwnProps {
     onDropDownItemMouseOver?: (ref: ObjRefInScope) => void;
```
Comment (anvuong119) [inline libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdownBody.tsx:37]: Please update the copyright of this file as well. As we are making some changes now, the copyright should change accordingly, shouldn't it?

`// (C) 2020 GoodData Corporation`

↓

`// (C) 2020-2022 GoodData Corporation`

File: libs/sdk-ui-kit/styles/scss/SVGicons.scss [hunk 8]
```diff
@@ -41,6 +41,22 @@ $base_path: "~@gooddata/sdk-ui-kit/styles";
     }
 }
```
Comment (anvuong119) [inline libs/sdk-ui-kit/styles/scss/SVGicons.scss:43]: Please update the copyright of this file as well. As we are making some changes now, the copyright should change accordingly, shouldn't it?

`// (C) 2007-2020 GoodData Corporation`

↓

`// (C) 2007-2022 GoodData Corporation`

Comment (dlcd2203) [reply libs/sdk-ui-kit/styles/scss/SVGicons.scss:43]: fixed

File: libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdownBody.tsx (hunk 7 above)
Comment (dlcd2203) [reply libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdownBody.tsx:37]: fixed

File: libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdown.tsx (hunk 6 above)
Comment (dlcd2203) [reply libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdown.tsx:40]: fixed

File: libs/sdk-ui-filters/src/RankingFilter/RankingFilter.tsx (hunk 5 above)
Comment (dlcd2203) [reply libs/sdk-ui-filters/src/RankingFilter/RankingFilter.tsx:20]: fixed

File: libs/sdk-ui-filters/src/RankingFilter/MeasureDropdown/MeasureDropdownItem.tsx (hunk 4 above)
Comment (dlcd2203) [reply libs/sdk-ui-filters/src/RankingFilter/MeasureDropdown/MeasureDropdownItem.tsx:11]: fixed

File: libs/sdk-ui-filters/src/RankingFilter/MeasureDropdown/MeasureDropdownBody.tsx (hunk 3 above)
Comment (dlcd2203) [reply libs/sdk-ui-filters/src/RankingFilter/MeasureDropdown/MeasureDropdownBody.tsx:12]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6344/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8150/ : SUCCESS in 18m 44s

**Comment by anvuong119** [comment]
LGTM

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2089

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1837/ : SUCCESS in 1m 07s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2118/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/919/ : SUCCESS in 16m 25s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8154/ : SUCCESS in 16m 50s

**Pull request merged**